File: packages/shared/src/types/roomIdea.ts
/** A single note posted to a group practice room's shared idea board,
 * scoped to one question within the room. */
export interface RoomIdea {
  id: string;
  roomId: string;
  questionId: string;
  authorUserId: string;
  authorName: string;
  text: string;
  createdAt: string;
}

File: apps/backend/models/room_idea.py
"""Mirrors packages/server/src/models/RoomIdea.model.ts — same "roomideas"
collection. One shared-idea-board note, posted against one question within
one practice room."""

from __future__ import annotations

from datetime import UTC, datetime

from beanie import Document, PydanticObjectId
from pydantic import Field
from pymongo import IndexModel


class RoomIdea(Document):
    roomId: PydanticObjectId
    questionId: str
    authorUserId: PydanticObjectId
    authorName: str
    text: str
    createdAt: datetime = Field(default_factory=lambda: datetime.now(UTC))

    class Settings:
        name = "roomideas"
        indexes = [IndexModel([("roomId", 1), ("questionId", 1), ("createdAt", 1)])]
